plugins/glustershd/glustershd.go

```diff
@@ -62,7 +60,12 @@ func (shd *Glustershd) Args() string {
 
 // SocketFile returns path to the socket file used for IPC.
 func (shd *Glustershd) SocketFile() string {
-	return ""
+	if shd.socketfilepath != "" {
```

Comment:
`SocketFile()` is for location for socket file that self-heal daemon **listens on**. Is self-heal daemon listening on a socket file ? Does glusterd2 communicates with self heal daemon over socket file.

I think you're getting confused between listening on socket file and connecting to glusrerd's socket file.

Reply:
Move this code from SocketFile() to Args()